File: visitor/CalculateBonusVisitor.java
package visitor;

public class CalculateBonusVisitor implements ScheduleManagementVisitor {
    @Override
    public void visit(DayShiftScheduleManagement dayShiftScheduleManagement) {
        System.out.println("Calculating bonus for day shift...");
    }

    @Override
    public void visit(NightShiftScheduleManagement nightShiftScheduleManagement) {
        System.out.println("Calculating bonus for night shift...");
    }

    @Override
    public void visit(RemoteWorkShiftScheduleManagement remoteWorkShiftScheduleManagement) {
        System.out.println("Calculating bonus for remote work shift...");
    }
}

File: PHP/Patterns/src/Behavioral/Visitor/Visitors/CalculateBonusVisitor.php
<?php

namespace src\Behavioral\Visitor\Visitors;

use src\Behavioral\Visitor\DayShiftScheduleManagement;
use src\Behavioral\Visitor\NightShiftScheduleManagement;
use src\Behavioral\Visitor\RemoteWorkShiftScheduleManagement;

class CalculateBonusVisitor implements ScheduleManagementVisitor
{

    function visitDayShift(DayShiftScheduleManagement $dayShiftScheduleManagement)
    {
        print_r("Calculating Bonus for day Shift");
    }

    function visitNightShift(NightShiftScheduleManagement $nightShiftScheduleManagement)
    {
        print_r('Calculating Bonus for night Shift');
    }

    function visitRemoteWorkShift(RemoteWorkShiftScheduleManagement $remoteWorkShiftScheduleManagement)
    {
        print_r('Calculating Bonus for remote Work Shift');
    }
}